src/test/java/io/confluent/examples/streams/kafka/EmbeddedSingleNodeKafkaCluster.java

```diff
@@ -76,11 +78,25 @@ public void start() throws Exception {
     log.debug("Kafka instance is running at {}, connected to ZooKeeper at {}",
         broker.brokerList(), broker.zookeeperConnect());
 
-    schemaRegistry = new RestApp(
-        InstanceSpec.getRandomPort(),
-        zookeeperConnect(),
-        KAFKA_SCHEMAS_TOPIC, AVRO_COMPATIBILITY_TYPE);
-    schemaRegistry.start();
+    RestApp schemaRegistry = null;
+    int retries = 5;
+    while (retries > 0) {
+      schemaRegistry = new RestApp(
+          InstanceSpec.getRandomPort(),
+          zookeeperConnect(),
+          KAFKA_SCHEMAS_TOPIC, AVRO_COMPATIBILITY_TYPE);
+      try {
+        schemaRegistry.start();
+        break;
+      } catch (final BindException exception) {
```

Comment:
We once had a similar issue for the port-in-use root cause in AK: https://issues.apache.org/jira/browse/KAFKA-1501

Instead of retrying, I think a better approach was done there, is to "allow the port to be 0 to make the kernel give us a random port": https://reviews.apache.org/r/31806/diff/3#1